main/www/security.go

```diff
@@ -54,7 +54,7 @@ func redirectToLogin(c echo.Context) error {
 	if isAjax {
 		return echo.ErrUnauthorized
 	}
-	return c.Redirect(http.StatusTemporaryRedirect, "/login")
+	return c.Redirect(http.StatusTemporaryRedirect, "/login?redirect=" + req.URL.Path)
```

Comment:
Two issues with this:

- If there is no URL Path, or it's just the login itself, then we should not have a `?redirect=` parameter at all.
- The URL Path should be sanitized, i.e. only allow end-points on the same server to combat phishing attacks.